src/hotspot/share/opto/superword.cpp

```diff
@@ -4200,6 +4200,25 @@ SWPointer::SWPointer(MemNode* mem, SuperWord* slp, Node_Stack *nstack, bool anal
   NOT_PRODUCT(if(_slp->is_trace_alignment()) _tracer.restore_depth();)
   NOT_PRODUCT(_tracer.ctor_6(mem);)
 
+  // In the pointer analysis, and especially the AlignVector, analysis we assume that
+  // stride and scale are not too large. For example, we multiply "scale * stride",
+  // and assume that this does not overflow the int range. We also take "abs(scale)"
+  // and "abs(stride)", which would overflow for min_int = -(2^31). Still, we want
+  // to at least allow small and moderately large stride and scale. Therefore, we
+  // allow values up to 2^30, which is only a factor 2 smaller than the max/min int.
+  // Normal performance relevant code will have much lower values. And the restriction
+  // allows us to keep the rest of the autovectorization code much simpler, since we
+  // do not have to deal with overflows.
+  jlong long_scale  = _scale;
+  jlong long_stride = slp->lp()->stride_is_con() ? slp->iv_stride() : 0;
```

Comment:
Hmm. But what happens if the stride is now zero? Does that at least get caught later?
Or should we just return here, and make sure the SWPointer is not valid, so nothing bad can happen down the road?

Reply:
AFAICS, if we set stride = 0 here, the check would just pass without anything else happening. This turns this change into effectively no-op for non-constant strides.

I am thinking that is a cleaner/safer choice for 21u: we only protect from bad _constant_ strides, letting the behavior with non-constant strides the same. I don't think we should bail on non-constant strides here, since that might introduce performance regressions. This is also more in line with mainline: we don't bail on non-constant strides in mainline -- somehow managing to (almost) never be exposed to them. If/when we discover another bug in mainline that exposes to non-constant strides, we would need to fix it in downstream code.